build/Makefile

```diff
@@ -70,17 +67,20 @@ contgen: contgen.c
 	cc $< -g -o $@
 
 continuation_templates.h: contgen
-	$(BUILD)/contgen 10 10 > continuation_templates.h
+	./contgen 10 10 > continuation_templates.h
 
 lutf8lib.o: lutf8lib.c
 	cc -std=c99 -c -g -I$(LUAJIT_INCLUDES) $^
 
 bundle.o: bundle.c
 	cc -std=c99 -c -g -I$(LUAJIT_INCLUDES) $^
 
-
-%.js: $(BUILD)/../jssrc/%.ts
-	cd .. && tsc --pretty
+%.js: ../jssrc/%.ts
+	@# check if there is an appropriate typescript installation
+	@# otherwise install it locally with npm
+	@$(WITH_NPM_PATH) tsc -v | grep -q 'Version 2' || \
+		npm install typescript || (echo 'Please install npm' && exit 1)
```

Comment:
If you want to update the PR to do that I can throw together a package.json, or you're welcome to generate a skeletal one with typescript@2.0